server/app/Icepeak/Main.hs

```diff
@@ -77,33 +74,54 @@ runCore core = do
   httpServer <- HttpServer.new core
   let wsServer = WebsocketServer.acceptConnection core
 
+  let tryWithTag wrapper act = catch (act >> pure RCR_Fine) $ \exc -> case fromException exc of
+        Just (_ :: AsyncException) -> pure $ RCR_Fine
+        _ -> pure $ wrapper exc
+
   -- start threads
-  pops <- Async.async $ Core.runCommandLoop core
+  pops <- Async.async $ tryWithTag RCR_CommandLoop $ Core.runCommandLoop core
+  upds <- Async.async $ tryWithTag RCR_WebSockets $ WebsocketServer.processUpdates core
+  serv <- Async.async $ tryWithTag RCR_HTTP $ Server.runServer logger wsServer httpServer
   sync <- Async.async $ Core.runSyncTimer core
```

Comment:
Let's rename these, I just had a hard time remembering what `pops` stands for again..
E.g. `coreThread`, `websocketThread`, `httpThread`, `syncThread`. I think that's clearer.

Reply:
Much nicer indeed